### DISCUSSION TITLE
Search for file by name or pattern anywhere in repository

### DISCUSSION THREAD

**Discussion by vsoch** [community/community#9710] [Other Feature Feedback, Questions, & Ideas] [answered]
I don't see that the general use case for search for files across a repository is supported.

I would have wanted to do: `file:Singularity` to look for a file named "Singularity" anywhere in a repository, or `file:Singularity.*` for the same with any extension, or even some other means to ask for an optional extension.

**Comment by look** [comment, marked as answer]
Thanks for the feedback. At this time we only have qualifiers for searching extensions and paths, not file names specifically. Both support regular expressions, so you could do something like [`path:/\/singularity/`](https://cs.github.com/?scopeName=All+repos&scope=&q=path%3A%2F%5C%2Fsingularity%2F).
